libraries/ms-helper/inc/can_interval.h

```diff
@@ -0,0 +1,39 @@
+#pragma once
+// Module for periodically sending CAN messages.
+// Requires a generic can module, soft_timers and interrupts to be enabled.
+
+#include <stdint.h>
+
+#include "generic_can.h"
+#include "generic_can_msg.h"
+#include "soft_timer.h"
+#include "status.h"
+
+#define CAN_INTERVAL_POOL_SIZE 5
+
+typedef struct CanInterval {
+  GenericCan *can;
+  GenericCanMsg *msg;
+  SoftTimerID timer_id;
+  uint32_t period;
+} CanInterval;
+
+// Initializes the can interval module.
+void can_interval_init(void);
+
+// Returns |interval| a pointer to a CANInterval object which stores the provided settings.
+StatusCode can_interval_factory(const GenericCan *can, const GenericCanMsg *msg, uint32_t period,
```

Comment:
Should mention that `msg` should persist or store it within CanInterval?

Reply:
Done. I decided for convenience that persisting the message in the object is probably worth the memory cost.